### ISSUE TITLE
Plugin colliding with each other when using Facebook SDK.

### ISSUE THREAD

**Issue by RaymondMoay** [googleads/googleads-mobile-unity#401]
Hi, when building for android in unity, the console gives out this error: 

"Found plugins with same names and architectures, Assets/Plugins/Android/libs/appcompat-v7-23.4.0.aar () and Assets/FacebookSDK/Plugins/Android/libs/appcompat-v7-23.4.0.aar (). Assign different architectures or delete the duplicate."

I tried deleting one of it, but that leads to even more errors. How should I proceed?

Thank you.

**Comment by RaymondMoay** [comment]
Alright so I reimported AdMob and now it seems to build fine. On my second try to build it out to android however, I get a "Unable to convert classes into dex format" error:

CommandInvokationFailure: Unable to convert classes into dex format.
/Library/Java/JavaVirtualMachines/jdk1.8.0_101.jdk/Contents/Home/bin/java -Xmx2048M -Dcom.android.sdkmanager.toolsdir="/Users/Raymond/Work/App Development/Android/sdk/tools" -Dfile.encoding=UTF8 -jar "/Applications/Unity/PlaybackEngines/AndroidPlayer/Tools/sdktools.jar" -

stderr[
Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/v4/BuildConfig;
Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/v4/accessibilityservice/AccessibilityServiceInfoCompat;
Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/v4/accessibilityservice/AccessibilityServiceInfoCompat$AccessibilityServiceInfoIcsImpl;
Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/v4/accessibilityservice/AccessibilityServiceInfoCompat$AccessibilityServiceInfoJellyBeanMr2;
Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/v4/accessibilityservice/AccessibilityServiceInfoCompat$AccessibilityServiceInfoStubImpl;
Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/v4/accessibilityservice/AccessibilityServiceInfoCompat$AccessibilityServiceInfoVersionImpl;
Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/v4/animation/AnimatorCompatHelper;
Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/v4/app/ActionBarDrawerToggle;
Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/v4/app/ActionBarDrawerToggle$1;

UNEXPECTED TOP-LEVEL EXCEPTION:
java.lang.RuntimeException: Translation has been interrupted
	at com.android.dx.command.dexer.Main.processAllFiles(Main.java:613)
	at com.android.dx.command.dexer.Main.runMonoDex(Main.java:311)
	at com.android.dx.command.dexer.Main.run(Main.java:277)
	at com.android.dx.command.dexer.Main.main(Main.java:245)
	at com.android.dx.command.Main.main(Main.java:106)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at SDKMain.main(SDKMain.java:129)
Caused by: java.lang.InterruptedException: Too many errors
	at com.android.dx.command.dexer.Main.processAllFiles(Main.java:605)
	... 9 more
]
stdout[
processing archive /Users/Raymond/Work/App Development/Projects/Jumpy/Temp/StagingArea/android-libraries/GoogleAIDL/libs/./classes.jar...
processing com/android/vending/billing/BuildConfig.class...
processing com/android/vending/billing/IInAppBillingService.class...
processing com/android/vending/billing/IInAppBillingService$Stub.class...
processing com/android/vending/billing/IInAppBillingService$Stub$Proxy.class...
processing archive /Users/Raymond/Work/App Development/Projects/Jumpy/Temp/StagingArea/android-libraries/GoogleMobileAdsPlugin/libs/./unity-plugin-library.jar...
processing com/google/unity/ads/AdNetworkExtras.class...
processing com/google/unity/ads/Banner.class...
processing com/google/unity/ads/Banner$1.class...
processing com/google/unity/ads/Banner$2.class...
processing com/google/unity/ads/Banner$3.class...
processing com/google/unity/ads/Banner$4.class...
processing com/google/unity/ads/Banner$5.class...
processing com/google/unity/ads/Banner$6.class...
processing com/google/unity/ads/Banner$7.class...
processing com/google/unity/ads/BuildConfig.class...
processing com/google/unity/ads/CustomNativeAd.class...
processing com/google/unity/ads/CustomNativeAd$1.class...
processing com/google/unity/ads/CustomNativeAd$2.class...
processing com/google/unity/ads/Interstitial.class...
processing com/google/unity/ads/Interstitial$1.class...
processing com/google/unity/ads/Interstitial$1$1.class...
processing com/google/unity/ads/Interstitial$2.class...
processing com/google/unity/ads/Interstitial$3.class...
processing com/google/unity/ads/NativeAdLoader.class...
processing com/google/unity/ads/NativeAdLoader$1.class...
processing com/google/unity/ads/NativeAdLoader$2.class...
processing com/google/unity/ads/NativeAdLoader$3.class...
processing com/google/unity/ads/NativeAdLoader$3$1.class...
processing com/google/unity/ads/NativeAdLoader$3$2.class...
processing com/google/unity/ads/NativeAdLoader$3$3.class...
processing com/google/unity/ads/NativeAdLoader$4.class...
processing com/google/unity/ads/NativeExpressAd.class...
processing com/google/unity/ads/NativeExpressAd$1.class...
processing com/google/unity/ads/NativeExpressAd$2.class...
processing com/google/unity/ads/NativeExpressAd$3.class...
processing com/google/unity/ads/NativeExpressAd$4.class...
processing com/google/unity/ads/NativeExpressAd$5.class...
processing com/google/unity/ads/NativeExpressAd$6.class...
processing com/google/unity/ads/NativeExpressAd$7.class...
processing com/google/unity/ads/NativeExpressAd$8.class...
processing com/google/unity/ads/PluginUtils.class...
processing com/google/unity/ads/RewardBasedVideo.class...
processing com/google/unity/ads/RewardBasedVideo$1.class...
processing com/google/unity/ads/RewardBasedVideo$1$1.class...
processing com/google/unity/ads/RewardBasedVideo$2.class...
processing com/google/unity/ads/RewardBasedVideo$3.class...
processing com/google/unity/ads/UnityAdListener.class...
processing com/google/unity/ads/UnityCustomNativeAdListener.class...
processing com/google/unity/ads/UnityRewardBasedVideoAdListener.class...
processing archive /Users/Raymond/Work/App Development/Projects/Jumpy/Temp/StagingArea/android-libraries/GooglePlay/libs/./classes.jar...
processing com/unity/purchasing/googleplay/ActivityLauncher.class...
processing com/unity/purchasing/googleplay/BillingServiceManager.class...
processing com/unity/purchasing/googleplay/BillingServiceManager$1.class...
processing com/unity/purchasing/googleplay/BillingServiceManager$1$1.class...
processing com/unity/purchasing/googleplay/BillingServiceManager$1$2.class...
processing com/unity/purchasing/googleplay/BillingServiceManager$2.class...
processing com/unity/purchasing/googleplay/BillingServiceProcessor.class...
processing com/unity/purchasing/googleplay/BuildConfig.class...
processing com/unity/purchasing/googleplay/Consts.class...
processing com/unity/purchasing/googleplay/Consts$PurchaseState.class...
processing com/unity/purchasing/googleplay/Consts$ResponseCode.class...
processing com/unity/purchasing/googleplay/GooglePlayBillingUnAvailableException.class...
processing com/unity/purchasing/googleplay/GooglePlayPurchasing.class...
processing com/unity/purchasing/googleplay/GooglePlayPurchasing$1.class...
processing com/unity/purchasing/googleplay/GooglePlayPurchasing$2.class...
processing com/unity/purchasing/googleplay/GooglePlayPurchasing$3.class...
processing com/unity/purchasing/googleplay/GooglePlayPurchasing$4.class...
processing com/unity/purchasing/googleplay/GooglePlayPurchasing$5.class...
processing com/unity/purchasing/googleplay/GooglePlayPurchasing$6.class...
processing com/unity/purchasing/googleplay/GooglePlayPurchasing$7.class...
processing com/unity/purchasing/googleplay/IActivityLauncher.class...
processing com/unity/purchasing/googleplay/IBillingServiceManager.class...
processing com/unity/purchasing/googleplay/IabException.class...
processing com/unity/purchasing/googleplay/IabHelper.class...
processing com/unity/purchasing/googleplay/IabHelper$1.class...
processing com/unity/purchasing/googleplay/IabHelper$2.class...
processing com/unity/purchasing/googleplay/IabHelper$2$1.class...
processing com/unity/purchasing/googleplay/IabHelper$3.class...
processing com/unity/purchasing/googleplay/IabHelper$4.class...
processing com/unity/purchasing/googleplay/IabHelper$OnConsumeFinishedListener.class...
processing com/unity/purchasing/googleplay/IabHelper$OnConsumeMultiFinishedListener.class...
processing com/unity/purchasing/googleplay/IabHelper$OnIabPurchaseFinishedListener.class...
processing com/unity/purchasing/googleplay/IabHelper$OnIabSetupFinishedListener.class...
processing com/unity/purchasing/googleplay/IabHelper$QueryInventoryFinishedListener.class...
processing com/unity/purchasing/googleplay/IabResult.class...
processing com/unity/purchasing/googleplay/Inventory.class...
processing com/unity/purchasing/googleplay/Purchase.class...
processing com/unity/purchasing/googleplay/PurchaseActivity.class...
processing com/unity/purchasing/googleplay/PurchaseFailedEvent.class...
processing com/unity/purchasing/googleplay/SkuDetails.class...
processing archive /Users/Raymond/Work/App Development/Projects/Jumpy/Temp/StagingArea/android-libraries/animated-vector-drawable-23.4.0/libs/./classes.jar...
processing android/support/graphics/drawable/AnimatedVectorDrawableCompat.class...
processing android/support/graphics/drawable/AnimatedVectorDrawableCompat$1.class...
processing android/support/graphics/drawable/AnimatedVectorDrawableCompat$AnimatedVectorDrawableCompatState.class...
processing android/support/graphics/drawable/AnimatedVectorDrawableCompat$AnimatedVectorDrawableDelegateState.class...
processing android/support/graphics/drawable/animated/BuildConfig.class...
processing archive /Users/Raymond/Work/App Development/Projects/Jumpy/Temp/StagingArea/android-libraries/appcompat-v7-23.4.0/libs/./classes.jar...
ignored resource .readme
processing android/support/v7/app/ActionBar.class...
processing android/support/v7/app/ActionBar$DisplayOptions.class...
processing android/support/v7/app/ActionBar$LayoutParams.class...
processing android/support/v7/app/ActionBar$NavigationMode.class...
processing android/support/v7/app/ActionBar$OnMenuVisibilityListener.class...
processing android/support/v7/app/ActionBar$OnNavigationListener.class...
processing android/support/v7/app/ActionBar$Tab.class...
processing android/support/v7/app/ActionBar$TabListener.class...
processing android/support/v7/app/ActionBarActivity.class...
processing android/support/v7/app/ActionBarDrawerToggle.class...
processing android/support/v7/app/ActionBarDrawerToggle$1.class...
processing android/support/v7/app/ActionBarDrawerToggle$Delegate.class...
processing android/support/v7/app/ActionBarDrawerToggle$DelegateProvider.class...
processing android/support/v7/app/ActionBarDrawerToggle$DrawerArrowDrawableToggle.class...
processing android/support/v7/app/ActionBarDrawerToggle$DrawerToggle.class...
processing android/support/v7/app/ActionBarDrawerToggle$DummyDelegate.class...
processing android/support/v7/app/ActionBarDrawerToggle$HoneycombDelegate.class...
processing android/support/v7/app/ActionBarDrawerToggle$JellybeanMr2Delegate.class...
processing android/support/v7/app/ActionBarDrawerToggle$ToolbarCompatDelegate.class...
processing android/support/v7/app/ActionBarDrawerToggleHoneycomb.class...
processing android/support/v7/app/ActionBarDrawerToggleHoneycomb$SetIndicatorInfo.class...
processing android/support/v7/app/AlertController.class...
processing android/support/v7/app/AlertController$1.class...
processing android/support/v7/app/AlertController$2.class...
processing android/support/v7/app/AlertController$3.class...
processing android/support/v7/app/AlertController$4.class...
processing android/support/v7/app/AlertController$5.class...
processing android/support/v7/app/AlertController$AlertParams.class...
processing android/support/v7/app/AlertController$AlertParams$1.class...
processing android/support/v7/app/AlertController$AlertParams$2.class...
processing android/support/v7/app/AlertController$AlertParams$3.class...
processing android/support/v7/app/AlertController$AlertParams$4.class...
processing android/support/v7/app/AlertController$AlertParams$OnPrepareListViewListener.class...
processing android/support/v7/app/AlertController$ButtonHandler.class...
processing android/support/v7/app/AlertController$CheckedItemAdapter.class...
processing android/support/v7/app/AlertDialog.class...
processing android/support/v7/app/AlertDialog$Builder.class...
processing android/support/v7/app/AppCompatActivity.class...
processing android/support/v7/app/AppCompatCallback.class...
processing android/support/v7/app/AppCompatDelegate.class...
processing android/support/v7/app/AppCompatDelegate$NightMode.class...
processing android/support/v7/app/AppCompatDelegateImplBase.class...
processing android/support/v7/app/AppCompatDelegateImplBase$1.class...
processing android/support/v7/app/AppCompatDelegateImplBase$ActionBarDrawableToggleImpl.class...
processing android/support/v7/app/AppCompatDelegateImplBase$AppCompatWindowCallbackBase.class...
processing android/support/v7/app/AppCompatDelegateImplV11.class...
processing android/support/v7/app/AppCompatDelegateImplV14.class...
processing android/support/v7/app/AppCompatDelegateImplV14$AppCompatWindowCallbackV14.class...
processing android/support/v7/app/AppCompatDelegateImplV23.class...
processing android/support/v7/app/AppCompatDelegateImplV23$AppCompatWindowCallbackV23.class...
processing android/support/v7/app/AppCompatDelegateImplV7.class...
processing android/support/v7/app/AppCompatDelegateImplV7$1.class...
processing android/support/v7/app/AppCompatDelegateImplV7$2.class...
processing android/support/v7/app/AppCompatDelegateImplV7$3.class...
processing android/support/v7/app/AppCompatDelegateImplV7$4.class...
processing android/support/v7/app/AppCompatDelegateImplV7$5.class...
processing android/support/v7/app/AppCompatDelegateImplV7$5$1.class...
processing android/support/v7/app/AppCompatDelegateImplV7$6.class...
processing android/support/v7/app/AppCompatDelegateImplV7$ActionMenuPresenterCallback.class...
processing android/support/v7/app/AppCompatDelegateImplV7$ActionModeCallbackWrapperV7.class...
processing android/support/v7/app/AppCompatDelegateImplV7$ActionModeCallbackWrapperV7$1.class...
processing android/support/v7/app/AppCompatDelegateImplV7$ListMenuDecorView.class...
processing android/support/v7/app/AppCompatDelegateImplV7$PanelFeatureState.class...
processing android/support/v7/app/AppCompatDelegateImplV7$PanelFeatureState$SavedState.class...
processing android/support/v7/app/AppCompatDelegateImplV7$PanelFeatureState$SavedState$1.class...
processing android/support/v7/app/AppCompatDelegateImplV7$PanelMenuPresenterCallback.class...
processing android/support/v7/app/AppCompatDialog.class...
processing android/support/v7/app/AppCompatDialogFragment.class...
processing android/support/v7/app/AppCompatViewInflater.class...
processing android/support/v7/app/AppCompatViewInflater$DeclaredOnClickListener.class...
processing android/support/v7/app/NavItemSelectedListener.class...
processing android/support/v7/app/NotificationCompat.class...
processing android/support/v7/app/NotificationCompat$1.class...
processing android/support/v7/app/NotificationCompat$Builder.class...
processing android/support/v7/app/NotificationCompat$IceCreamSandwichExtender.class...
processing android/support/v7/app/NotificationCompat$JellybeanExtender.class...
processing android/support/v7/app/NotificationCompat$LollipopExtender.class...
processing android/support/v7/app/NotificationCompat$MediaStyle.class...
processing android/support/v7/app/NotificationCompatImpl21.class...
processing android/support/v7/app/NotificationCompatImplBase.class...
processing android/support/v7/app/ToolbarActionBar.class...
processing android/support/v7/app/ToolbarActionBar$1.class...
processing android/support/v7/app/ToolbarActionBar$2.class...
processing android/support/v7/app/ToolbarActionBar$ActionMenuPresenterCallback.class...
processing android/support/v7/app/ToolbarActionBar$MenuBuilderCallback.class...
processing android/support/v7/app/ToolbarActionBar$PanelMenuPresenterCallback.class...
processing android/support/v7/app/ToolbarActionBar$ToolbarCallbackWrapper.class...
processing android/support/v7/app/TwilightCalculator.class...
processing android/support/v7/app/TwilightManager.class...
processing android/support/v7/app/TwilightManager$1.class...
processing android/support/v7/app/TwilightManager$TwilightState.class...
processing android/support/v7<message truncated>

How do I solve this?

**Comment by rampara** [comment]
@RaymondMoay it looks like you're including the v4 support library multiple times in your Unity project. Make sure the corresponding AAR for this library is only included once in your project.

**Comment by RaymondMoay** [comment]
@rampara Alright, I removed the support library and it works for android now. However, when built to IOS, there is an error: Module 'GoogleMobileAds' not found GADUNativeCustomTemplateAD.h in Xcode.

**Comment by rampara** [comment]
When building for iOS, do you see any errors in the Unity console?

**Comment by RaymondMoay** [comment]
Yes there are three outputs:

(1) Project GPM data does not exist. Creating new object.
UnityEngine.Debug:Log(Object)
Google.PackageManager.LoggingController:Log(String)
Google.PackageManager.ProjectManagerController:InflateProjectRecord(String)
Google.PackageManager.ProjectManagerController:.cctor()
UnityEditor.EditorAssemblies:ProcessInitializeOnLoadAttributes()

** Issues 2 and 3 are weird, because I do have POD installed on my Mac pro.

(2) 'pod' command not found; unable to generate a usable Xcode project. You can install cocoapods with the Ruby gem package manager:

(3) Error running cocoapods. Please ensure you have at least version 1.0.0.  You can install cocoapods with the Ruby gem package manager:

**Comment by RaymondMoay** [comment]
I have resolved the pod issue by reinstalling google ad mobs manually using Cocoapods on my Xcode project. However, the Project GPM data does not exist issue still persists in Unity.

**Comment by rampara** [comment]
@RaymondMoay is the "Project GPM data does not exist. Creating new object." message an or a warning? Are you still able to build and run your app?

**Comment by GcubeTech** [comment]
am facing same problem any solution for it?

**Comment by RaymondMoay** [comment]
@rampara It is a message, and it does not affect my build. I am still having issues with Cocoapod, however I found a solution. All I did was manually update pod and it works! However, I think this issue deserves to be fixed. 

@GcubeTech If your issue is regarding Cocoapods not installing from unity, update pod manually and it should work. If it's the Facebook plugin, just remove the extra v4 support libraries and it worked.
